src/dackar/utils/mbse/customMBSEparser.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2024, Battelle Energy Alliance, LLC  ALL RIGHTS RESERVED
+
+"""
+Created on February, 2024
+
+@author: mandd
+"""
+
+# External Imports
+import pandas as pd
+
+class customMBSEobject(object):
+    """
+        Class designed to process the a custom MBSE model from file.
+    """
+    def __init__(self, nodes_filename, edges_filename):
+        """
+        Initialization method for the custom MBSE model class
+
+        Args:
+
+            nodes_filename: file, file in .csv format containing all nodes 
+            edges_filename: file, file in .csv format containing all edges 
+
+        Returns:
+
+            None
+        """
+        self.nodes_filename = nodes_filename
+        self.edges_filename = edges_filename
+        self.entities = []
+        self.listIDs = []
+
+        self.allowed_node_types = ['entity']
+        self.allowed_edge_types = ['link','composition','support'] # to be developed: 'opm_instance'
+
+        self.allowed_node_cols = ['nodeID','label','ID','type','description']
+        self.allowed_edge_cols = ['sourceNodeId','targetNodeId','type','medium','description']
+
+        self.parseFiles()
+        self.checkNodesFile()
+        self.checkEdgesFile()
+    
+    def parseFiles(self):
+        """
+        Medthods designed to parse the node and edge files
+
+        Args:
+
+            None
+
+        Returns:
+
+            None
+        """
+        # parse nodes
+        self.nodes_df = pd.read_csv(self.nodes_filename, sep=',', skip_blank_lines=True, dtype=str)
+        self.nodes_df.dropna(how='all', inplace=True)
+
+        self.entities = self.nodes_df['nodeID'].to_list()
+        # parse edges
+        self.edges_df = pd.read_csv(self.edges_filename, sep=',', skip_blank_lines=True, dtype=str)
+        self.edges_df.dropna(how='all', inplace=True)
+
+    def checkNodesFile(self):
+        """
+        Medthods designed to check the node file
+
+        Args:
+
+            None
+
+        Returns:
+
+            None
+        """
+        print('- Check node file -')
+        # Check all columns are present
+        cols = self.nodes_df.columns.tolist()
+        if set(cols)!=set(self.allowed_node_cols):
+            raise IOError('Node file structure check - Error: wrong set of provided columns (allowed: nodeID, label, ID, type)')
+        else:
+            print('Node file structure check - Pass')
+
+        # Check for duplicate IDs
+        duplicateIDs = self.nodes_df.duplicated()
+
+        if self.nodes_df[duplicateIDs].empty:
+            print("List of node IDs check - Pass")
```

Comment:
I would recommend to use logging instead of print. You can find examples in DACKAR/src/dackar/workflows/WorkflowBase.py

Reply:
Fixed